senders/metrics_expirer.go

```diff
@@ -0,0 +1,88 @@
+package senders
+
+import (
+	"sync"
+	"time"
+
+	"github.com/mitchellh/hashstructure"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricID struct {
+	Name   string
+	Labels prometheus.Labels
+}
+
+type MetricsExpirer struct {
+	mtx              sync.Mutex
+	metricIDs        map[uint64]metricID
```

Comment:
It might be neater to have another struct here instead of matching on keys. It would make the pairing of the two pieces of data more explicit.
```
type MetricActivity struct {
    MetricID metricID
    Timestamp time.Time
}
```
Then this line and line below would be replaced with something like:
`metricActivities map[uint64]MetricActivity`

Reply:
Yep, you are right, I didn't due pure laziness :angel: 

I will change it

Reply:
I pushed a commit. Check it out. It requires a `git rebase -i master --autosquash`.